google/cloud/sql/connector/InstanceConnectionManager.py

```diff
@@ -38,12 +37,43 @@
 
 
 class InstanceMetadata:
-    ip_addresses: Dict[str, str]
-    ssl_context: OpenSSL.SSL.Context
+    ip_address: str
+    ca_fileobject: NamedTemporaryFile
+    cert_fileobject: NamedTemporaryFile
+    key_fileobject: NamedTemporaryFile
+    context: ssl.SSLContext
 
-    def __init__(self, ssl_context, ip_addresses):
-        self.ssl_context = ssl_context
-        self.ip_addresses = ip_addresses
+    def __init__(
+        self,
+        ephemeral_cert: str,
+        ip_address: str,
+        private_key: str,
+        server_ca_cert: str,
+    ):
+        self.ip_address = ip_address
+
+        self.ca_fileobject = NamedTemporaryFile(suffix=".pem")
```

Comment:
The docs for `NamedTemporaryFiles` warn that they cannot be opened again on Windows. https://docs.python.org/3/library/tempfile.html#tempfile.NamedTemporaryFile

The later lines concern me (`load_verify_locations`) as I am pretty sure that is going to attempt to open the file again.

Reply:
General windows incompatibility is being tracked in #22